services/workflows-service/src/events/document-changed-webhook-caller.ts

```diff
@@ -75,48 +80,69 @@ export class DocumentChangedWebhookCaller {
       return;
     }
 
-    const { id, environment, url, authSecret, apiVersion } = getWebhookInfo(
+    const webhooks = getWebhooks(
       data.updatedRuntimeData.config,
       this.configService.get('NODE_ENV'),
-      this.configService.get('WEBHOOK_URL'),
-      this.configService.get('WEBHOOK_SECRET'),
       'workflow.context.document.changed',
     );
 
-    if (!url) {
-      this.logger.log(`No webhook url found for a workflow runtime data with an id of "${id}"`);
-      return;
-    }
-
     data.updatedRuntimeData.context.documents.forEach((doc: any) => {
       delete doc.propertiesSchema;
     });
 
+    const customer = await this.customerService.getByProjectId(data.updatedRuntimeData.projectId);
+
+    const { webhookSharedSecret } =
+      customer.authenticationConfiguration as TAuthenticationConfiguration;
+
+    for (const webhook of webhooks) {
```

Comment:
Any reason not to use `Promise.all([...])` like this:
```
const webhookPromises = webhooks.map((webhook) => 
    this.sendWebhook({ data, newDocumentsByIdentifier, oldDocuments, webhook, webhookSharedSecret })
);

await Promise.all(webhookPromises);
```
This way all webhooks will fire concurrently